src/ucp/api/ucp.h

```diff
@@ -1359,6 +1372,28 @@ ssize_t ucp_stream_worker_poll(ucp_worker_h worker,
                                ucp_stream_poll_ep_t *poll_eps, size_t max_eps,
                                unsigned flags);
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Add user defined callback for active message.
+ *
+ * This routine installs a user defined callback to handle incoming active
+ * messages with a specific id. This callback is called whenever an active message,
+ * which was sent from the remote peer by @ref for ucp_am_send_nb, is received on 
+ * this worker.
+ *
+ * @param [in]  worker      UCP worker on which to set the am handler
+ * @param [in]  id          Active message id.
+ * @param [in]  cb          Active message callback. NULL to clear.
+ * @param [in]  arg         Active message argument, which will be passed in to
+ *                          every invocation of the callback as the arg argument.
+ * @param [in]  flags       For future use
```

Comment:
@snyjm-18 pls add a flag which indicates exact behavior of the `cb` for forward compatibility.
For example, if your proposal assumes callback invoking on whole message only, it may be something like UCP_AM_FLAG_WHOLE_MSG and add to description that only this way is currently supported.